src/snyk/common/commands/commandController.ts

```diff
@@ -0,0 +1,106 @@
+import _ from 'lodash';
+import * as vscode from 'vscode';
+import { ISnykCodeService } from '../../snykCode/codeService';
+import { severityAsText } from '../../snykCode/utils/analysisUtils';
+import { createDCIgnore } from '../../snykCode/utils/ignoreFileUtils';
+import { CodeIssueCommandArg } from '../../snykCode/views/interfaces';
+import { OssService } from '../../snykOss/services/ossService';
+import { OssIssueCommandArg } from '../../snykOss/views/vulnerabilityProvider';
+import { analytics } from '../analytics/analytics';
+import {
+  SNYK_COPY_AUTH_LINK_COMMAND,
+  SNYK_OPEN_BROWSER_COMMAND,
+  SNYK_OPEN_LOCAL_COMMAND,
+  VSCODE_GO_TO_SETTINGS_COMMAND,
+} from '../constants/commands';
+import { COMMAND_DEBOUNCE_INTERVAL, IDE_NAME, SNYK_NAME_EXTENSION, SNYK_PUBLISHER } from '../constants/general';
+import { IOpenerService } from '../services/openerService';
+import { OpenIssueCommandArg, OpenCommandIssueType } from './types';
+
+export class CommandController {
+  private debouncedCommands: Record<string, _.DebouncedFunc<(...args: unknown[]) => Promise<unknown>>> = {};
+
+  constructor(
+    private openerService: IOpenerService,
+    private snykCode: ISnykCodeService,
+    private ossService: OssService,
+  ) {}
+
+  openBrowser(url: string): unknown {
+    return this.executeCommand(SNYK_OPEN_BROWSER_COMMAND, this.openerService.openBrowserUrl.bind(this), url);
+  }
+
+  copyAuthLink(): unknown {
+    return this.executeCommand(SNYK_COPY_AUTH_LINK_COMMAND, this.openerService.copyOpenedUrl.bind(this.openerService));
+  }
+
+  openLocal(path: vscode.Uri, range?: vscode.Range): void {
+    // todo: add error reporting
+    void vscode.window.showTextDocument(path, { viewColumn: vscode.ViewColumn.One, selection: range });
+  }
+
+  openSettings(): void {
+    void vscode.commands.executeCommand(VSCODE_GO_TO_SETTINGS_COMMAND, `@ext:${SNYK_PUBLISHER}.${SNYK_NAME_EXTENSION}`);
+  }
+
+  async createDCIgnore(custom = false, path?: string): Promise<void> {
+    if (!path) {
+      const paths = (vscode.workspace.workspaceFolders || []).map(f => f.uri.fsPath);
+      const promises = [];
+      for (const p of paths) {
+        promises.push(createDCIgnore(p, custom));
+      }
+      await Promise.all(promises);
+    } else {
+      await createDCIgnore(path, custom);
+    }
+  }
+
+  async openIssueCommand(arg: OpenIssueCommandArg): Promise<void> {
+    if (arg.issueType == OpenCommandIssueType.CodeIssue) {
+      const issue = arg.issue as CodeIssueCommandArg;
+      const suggestion = this.snykCode.analyzer.findSuggestion(issue.message);
+      if (!suggestion) return;
+      // Set openUri = null to avoid opening the file (e.g. in the ActionProvider)
+      if (issue.openUri !== null)
```

Comment:
any reason we're strongly checking against `null`?

Reply:
Opening an issue command is executed within editor, at which point user is looking at the opened file already. 
It's not a new code in this line, so I'd not touch it as part of this PR not to amend the existing Snyk Code open issue functionality.

Great attention to details though!